fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/transaction/Transaction.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.transaction;
+
+import net.fabricmc.fabric.impl.transfer.transaction.TransactionManagerImpl;
+
+/**
+ * A global operation where participants guarantee atomicity: either the whole operation succeeds,
+ * or it is completely aborted and rolled back.
+ *
+ * <p>Every transaction is only valid on the thread it was opened on,
+ * and attempts to call transaction functions on any other thread will throw an exception.
+ * Consequently, concurrent can be concurrent across multiple threads, as long as they don't share any state.
+ */
+public interface Transaction extends AutoCloseable {
+	/**
+	 * Open a new outer transaction.
```

Comment:
Maybe a small explanation of "outer transaction" would be useful here?

Reply:
Hopefully the new documentation helps.